emscripten-simulated-jerry-api/jerry.js

```diff
@@ -36,10 +36,13 @@
   };
 
   var hasInternalProps = function (value) {
-    return (canHaveInternalProps(value) && value.__jerryInternalProps);
+    return (canHaveInternalProps(value) && __jerry._jerryInternalPropsWeakMap.has(value));
   };
 
   global.__jerry = {
+    // Define a WeakMap to maintain the map between JS obj and its internal props
+    _jerryInternalPropsWeakMap: {},
```

Comment:
Should use `new WeakMap()`?

Reply:
already used the `new WeakMap()` in the reset